### ISSUE TITLE
Non-blocking initial load

### ISSUE THREAD

**Issue by merlinnot** [firebase/polymerfire#241]
Hi,

as far as I can see someone is working on speeding up initial load time https://github.com/firebase/polymerfire/pull/230. Unfortunately, there are some use cases where we'd like to show some data even if not all records are available, for example in some infinite-scrolling lists. The current implementation in `master` branch blocks the thread until all data is loaded. We have a prototype of a fairly simple solution to this problem using `setTimeout`.

I'd like to assign one of my employees to work on production-ready solution with extended documentation and tests, but I'd like it to be merged to official Polymerfire repo so we won't need to maintain our own fork merging all new releases of Polymerfire, since we do our best to use most up to date tools and libraries.

The question is: do you see a reason behind such option and can you guarantee, that if we provide the highest-quality solution to this problem, it will be merged into the official Polymerfire repo? If so, I'd also like to sign the Google's CLA on behalf of my company, [Codemachine](https://codemachine.studio). How can I do this? I'm a co-owner and a board member of the company.

**Comment by tjmonsi** [comment]
@merlinnot What you can do is just fork and do your solution, then do a PR. I don't know what's the timeline for the team but I do believe they are working on merging PRs that are really good.

As for what you have said, I agree that the data loaded blocks the thread. The good thing with loading a hundred units simultaneously using `on('value')` is it doesn't call the function for `on('child_added')` a hundred times (which blocks the main thread longer). With that said, data shows up instantaneously for a shorter period of time. 

As for doing what you have said, I feel that having a setTimeout on `on('child_added')` (to force it to only be called when the main thread is free) will make the user see that the data is being added one by one in the iron-list every time the user goes to that page (unless the iron-list has a different array to look at). 

A better strategy though is to use either Polymer.async and also add a flag on firebase-query (like `async` I guess) to say that you want the data to be loaded asynchronously

**Comment by mbleigh** [comment]
FYI I chatted with one of the Firebase DB SDK authors and he said that in general while `child_added` events are *fired* one by one, typically they don't start firing until the *entire node* has synced down and into memory. That means that the only perf benefit to "as they show up" would be if there's extremely heavy render cost. I think you can probably mitigate that in other ways.

**Comment by merlinnot** [comment]
If that's the case, it can be easily done without any modifications to polymerfire. Thanks for the feedback.
